code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,292 @@
+	if (!hoistee)
+		return
+	if (hoistee.z != source_hook.z)
+		release_hoistee()
+		return
+
+/obj/structure/hoist/proc/release_hoistee()
+	if(ismob(hoistee))
+		source_hook.unbuckle_mob(hoistee)
+	else
+		hoistee.anchored = FALSE
+	hoistee = null
+	layer = NORMAL_LAYER
+
+/obj/structure/hoist/proc/break_hoist()
+	if(broken)
+		return
+	broken = TRUE
+	desc += " It looks broken, and the clamp has retracted back into the hoist. Seems like you'd have to re-deploy it to get it to work again."
+	if(hoistee)
+		release_hoistee()
+	QDEL_NULL(source_hook)
+
+/obj/structure/hoist/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			if(prob(50))
+				qdel(src)
+			else
+				visible_message("\The [src] shakes violently, and neatly collapses as its damage sensors go off.")
+				collapse_kit()
+			return
+		if(3.0)
+			if(prob(50) && !broken)
+				break_hoist()
+			return
+
+/obj/effect/hoist_hook/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			source_hoist.break_hoist()
+			return
+		if(2.0)
+			if(prob(50))
+				source_hoist.break_hoist()
+			return
+		if(3.0)
+			if(prob(25))
+				source_hoist.break_hoist()
+			return
+
+
+/obj/structure/hoist/attack_hand(mob/living/user)
+	if (!ishuman(user))
+		return
+
+	if (user.incapacitated())
+		to_chat(user, SPAN_NOTICE("You can't do that while incapacitated."))
+		return
+
+	if (!user.check_dexterity(DEXTERITY_GRIP, silent = TRUE))
+		to_chat(user, SPAN_NOTICE("You stare cluelessly at \the [src]."))
+		return
+
+	if(broken)
+		to_chat(user, SPAN_WARNING("The hoist is broken!"))
+		return
+	var/can = can_move_dir(movedir)
+	var/movtext = movedir == UP ? "raise" : "lower"
+	if (!can) // If you can't...
+		movedir = movedir == UP ? DOWN : UP // switch directions!
+		to_chat(user, SPAN_NOTICE("You switch the direction of the pulley."))
+		return
+
+	if (!hoistee)
+		user.visible_message(
+			SPAN_NOTICE("[user] begins to [movtext] the clamp."),
+			SPAN_NOTICE("You begin to [movtext] the clamp."),
+			SPAN_NOTICE("You hear the sound of a crank."))
+		move_dir(movedir, 0)
+		return
+
+	check_consistency()
+
+	var/size
+	if (ismob(hoistee))
+		var/mob/M = hoistee
+		size = M.mob_size
+	else if (isobj(hoistee))
+		var/obj/O = hoistee
+		size = O.w_class
+
+	user.visible_message(
+		SPAN_NOTICE("[user] begins to [movtext] \the [hoistee]!"),
+		SPAN_NOTICE("You begin to [movtext] \the [hoistee]!"),
+		SPAN_NOTICE("You hear the sound of a crank."))
+	if (do_after(user, (1 SECONDS) * size / 4, src))
+		move_dir(movedir, 1)
+
+/obj/structure/hoist/proc/collapse_kit()
+	new /obj/item/hoist_kit(get_turf(src))
+	qdel(src)
+
+/obj/structure/hoist/verb/collapse_hoist()
+	set name = "Collapse Hoist"
+	set category = "Object"
+	set src in range(1)
+
+	if (!ishuman(usr))
+		return
+
+	if (isobserver(usr) || usr.incapacitated())
+		return
+	if (!usr.check_dexterity(DEXTERITY_GRIP, silent = TRUE)) // thanks nanacode
+		to_chat(usr, SPAN_NOTICE("You stare cluelessly at \the [src]."))
```

Comment:
This message isn't very informative about the actual reason the check is failing.

Reply:
Also, SPAN_WARNING for warning messages.